packages/loader/container-loader/src/containerStorageAdapter.ts

```diff
@@ -120,8 +128,20 @@ export class ContainerStorageAdapter
 			// A callback to ensure we fetch the most updated value of service.policies.summarizeProtocolTree, which could be set
 			// based on the response received from the service after connection is established.
 			() => {
-				this._summarizeProtocolTree =
-					this._summarizeProtocolTree ?? service.policies?.summarizeProtocolTree ?? false;
+				// If the enableSummarizeProtocolTree has disabled the single-commit summaries, then it is given a higher priority than driver policy.
+				// Driver policy is accounted for only when enableSummarizeProtocolTree was undefined or true.
+				const shouldSummarize =
+					this._summarizeProtocolTree === false
+						? false
+						: service.policies?.summarizeProtocolTree ?? false;
```

Comment:
Another way to write it (I think). Totally subjective, pick whichever is clearer to you.

```suggestion
				const shouldSummarize =
					this._summarizeProtocolTree !== false && service.policies?.summarizeProtocolTree === true;
```

Reply:
I'll add - it's usually easier to interpret the name of an optional boolean when it defaults to false, not true.  Consider how it would read if you refactored things to rename `_summarizeProtocolTree` to `_doNotSummarizeProtocolTree`, with default value false.